spring-messaging/src/test/java/org/springframework/messaging/rsocket/service/RSocketServiceIntegrationTests.java

```diff
@@ -124,6 +145,23 @@ interface Service {
 	}
 
 
+	@Controller
+	@RSocketExchange("exchange")
+	static class ExchangeServerController implements Service {
+
+		@Override
+		public Mono<String> echoAsync(String payload) {
+			return Mono.delay(Duration.ofMillis(10)).map(aLong -> payload + " async");
+		}
+
+		@Override
+		public Flux<String> echoStream(String payload) {
+			return Flux.interval(Duration.ofMillis(10)).map(aLong -> payload + " " + aLong);
+		}
+
+	}
```

Comment:
I don't think we need both `ServerController` and `ExchangeServerController`. `@MessageMapping` methods already have test coverage, and it is not a goal here to test that. We're instead testing requesting with `@RSocketExchange` methods. Now we can also test responding. 

I suggest keeping the test class as it was before, but simply replacing `@MessageMapping` with `@RSocketExchange`.

Reply:
I have removed the new controller and made the existing one implement the `Service` without annotating it directly; also have slightly modified the service to additionally verify class-level `@RSocketExchange`. Let me know if that looks good to you.